tests/test_metrics.py

```diff
@@ -0,0 +1,132 @@
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+import astropy.units as u
+import numpy as np
+import unittest
+
+import lsst.skymap as skyMap
+import lsst.utils.tests
+from lsst.verify import Name
+from lsst.verify.tasks.testUtils import MetricTaskTestCase
+
+from lsst.ap.pipe.createApFakes import CreateRandomApFakesTask, CreateRandomApFakesConfig
+from lsst.ap.pipe.metrics import ApFakesCompletenessMetricTask, ApFakesCompletenessMetricConfig
+
+
+class TestApCompletenessTask(MetricTaskTestCase):
+
+    @classmethod
+    def makeTask(cls, magMin=20, magMax=30):
+        """Make the task and allow for modification of the config min and max.
+
+        Parameters
+        ----------
+        magMin : min magnitude, `float`
+            Minimum magnitude
+        magMax : min magnitude, `float`
+            Maximum magnitude
+        """
+        config = ApFakesCompletenessMetricConfig()
+        config.magMin = magMin
+        config.magMax = magMax
+
+        return ApFakesCompletenessMetricTask(config=config)
+
+    def setUp(self):
+        super().setUp()
+
+        simpleMapConfig = skyMap.discreteSkyMap.DiscreteSkyMapConfig()
+        simpleMapConfig.raList = [45]
+        simpleMapConfig.decList = [45]
+        simpleMapConfig.radiusList = [0.1]
+
+        self.simpleMap = skyMap.DiscreteSkyMap(simpleMapConfig)
+        self.tractId = 0
+        bCircle = self.simpleMap.generateTract(self.tractId).getInnerSkyPolygon().getBoundingCircle()
+        self.targetSources = 1000
+        self.sourceDensity = (self.targetSources
+                              / (bCircle.getArea() * (180 / np.pi) ** 2))
+
+        fakesConfig = CreateRandomApFakesConfig()
+        fakesConfig.fraction = 0.0
+        fakesConfig.fakeDensity = self.sourceDensity
+        fakesTask = CreateRandomApFakesTask(config=fakesConfig)
+        fakeCat = fakesTask.run(self.tractId, self.simpleMap).fakeCat
+
+        self.band = 'g'
+        magCut = 25
+        magMask = (fakeCat[f"{fakesConfig.magVar}" % self.band] < magCut)
+        self.expectedAllMatched = magMask.sum()
+        ids = np.where(magMask, np.arange(1, len(fakeCat) + 1, dtype=int), 0)
+        self.fakeCat = fakeCat.assign(diaObjectId=ids,
+                                      filterName=["g"] * len(fakeCat),
+                                      diaSourceId=ids)
```

Comment:
This is a bit surprising. Shouldn't the catalog returned by `fakesTask` already have a filter column?

Reply:
The columns here are to mimic the results of matchFakes, not create fakes. The fakes have all magnitudes in all bands created. This is faking columns after processing.